File: ContractTests/DotNET/FilterParityCommandValidator.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Arc.Commands;
using FluentValidation;

namespace HttpFixture;

/// <summary>Rejects the fixture's invalid values.</summary>
public sealed class FilterParityCommandValidator : CommandValidator<FilterParityCommand>
{
    /// <summary>Declares the parity rule.</summary>
    public FilterParityCommandValidator()
    {
        RuleFor(command => command.Value).Must(value => !value.EndsWith("invalid", StringComparison.Ordinal))
            .WithMessage("Value is invalid");
    }
}

File: ContractTests/Http/modelBound/FilterParityCommandValidator.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.
import { CommandValidator, validator } from '@cratis/arc.core';
import { FilterParityCommand } from './FilterParityCommand.js';
import { recordFilterParity } from './FilterParityObservations.js';

/** Reject the fixture's invalid values. */
@validator(FilterParityCommand)
export class FilterParityCommandValidator extends CommandValidator<FilterParityCommand> {
    constructor() {
        super();
        recordFilterParity('command validator constructed');
        this.ruleFor(command => command.value).must(value => {
            recordFilterParity('command validator', value);
            return !value.endsWith('invalid');
        }).withMessage('Value is invalid');
    }
}
